tools/Az.Tools.Predictor/Az.Tools.Predictor/AzPredictorService.cs

```diff
@@ -143,22 +144,46 @@ protected virtual void Dispose(bool disposing)
         /// Tries to get the suggestions for the user input from the command history. If that doesn't find
         /// <paramref name="suggestionCount"/> suggestions, it'll fallback to find the suggestion regardless of command history.
         /// </remarks>
-        public CommandLineSuggestion GetSuggestion(Ast input, int suggestionCount, int maxAllowedCommandDuplicate, CancellationToken cancellationToken)
+        public CommandLineSuggestion GetSuggestion(PredictionContext context, int suggestionCount, int maxAllowedCommandDuplicate, CancellationToken cancellationToken)
         {
-            Validation.CheckArgument(input, $"{nameof(input)} cannot be null");
+            Validation.CheckArgument(context, $"{nameof(context)} cannot be null");
             Validation.CheckArgument<ArgumentOutOfRangeException>(suggestionCount > 0, $"{nameof(suggestionCount)} must be larger than 0.");
             Validation.CheckArgument<ArgumentOutOfRangeException>(maxAllowedCommandDuplicate > 0, $"{nameof(maxAllowedCommandDuplicate)} must be larger than 0.");
 
-            var commandAst = input.FindAll(p => p is CommandAst, true).LastOrDefault() as CommandAst;
-            var commandName = (commandAst?.CommandElements?.FirstOrDefault() as StringConstantExpressionAst)?.Value;
+            var relatedAsts = context.RelatedAsts;
+            CommandAst commandAst = null;
+
+            for (var i = relatedAsts.Count - 1; i >= 0; --i)
+            {
+                if (relatedAsts[i] is CommandAst c)
+                {
+                    commandAst = c;
+                    break;
+                }
+            }
+
+            var commandName = commandAst?.GetCommandName();
 
             if (string.IsNullOrWhiteSpace(commandName))
             {
                 return null;
             }
 
-            var inputParameterSet = new ParameterSet(commandAst);
-            var rawUserInput = input.Extent.Text;
+            ParameterSet inputParameterSet = null;
+
+            try
+            {
+                inputParameterSet = new ParameterSet(commandAst);
+            }
+            catch when (!IsSupportedCommand(commandName))
+            {
+                // We only ignore the exception when the command name is not supported.
+                // For the supported ones, this most likely happens when positional parameters are used.
+                // We want to collect the telemetry about the exception how common a positional parameter is used.
+                return null;
```

Comment:
Is there something besides null we can return here? Just don't like giving "others" null if it can be helped. If not, please add it to the method comments.

Reply:
I add to the method comments to indicate null is returned when the user input context isn't valid/supported. It's v.s. a default value (en empty collection) meaning that we try to find the suggestion but there are not.